timelock-impl/src/main/java/com/palantir/atlasdb/timelock/adjudicate/LeaderElectionMetricAggregator.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.adjudicate;
+
+import com.palantir.atlasdb.util.CurrentValueMetric;
+import com.palantir.atlasdb.util.MetricsManager;
+import com.palantir.atlasdb.util.SlidingWindowWeightedMeanGauge;
+import com.palantir.conjure.java.lib.SafeLong;
+import com.palantir.timelock.feedback.LeaderElectionStatistics;
+
+public final class LeaderElectionMetricAggregator {
+    private final SlidingWindowWeightedMeanGauge weightedGaugeP99;
+    private final SlidingWindowWeightedMeanGauge weightedGaugeP95;
+    private final SlidingWindowWeightedMeanGauge weightedGaugeMean;
+    private final CurrentValueMetric<Long> lastElectionTime;
+
+    public LeaderElectionMetricAggregator(MetricsManager metricsManager) {
+        weightedGaugeP99 = SlidingWindowWeightedMeanGauge.create();
+        weightedGaugeP95 = SlidingWindowWeightedMeanGauge.create();
+        weightedGaugeMean = SlidingWindowWeightedMeanGauge.create();
+        lastElectionTime = new CurrentValueMetric<>();
+        metricsManager.registerMetric(
+                LeaderElectionMetricAggregator.class, "leaderElectionImpactMean", weightedGaugeMean);
+        metricsManager.registerMetric(
+                LeaderElectionMetricAggregator.class, "leaderElectionImpactP95", weightedGaugeP95);
+        metricsManager.registerMetric(
+                LeaderElectionMetricAggregator.class, "leaderElectionImpactP99", weightedGaugeP99);
+        metricsManager.registerMetric(LeaderElectionMetricAggregator.class, "leaderElectionDuration", lastElectionTime);
+    }
+
+    void report(LeaderElectionStatistics statistics) {
+        long count = statistics.getCount().longValue();
+        weightedGaugeMean.update(statistics.getMean(), count);
+        weightedGaugeP95.update(statistics.getP95(), count);
+        weightedGaugeP99.update(statistics.getP99(), count);
+        lastElectionTime.setValue(
```

Comment:
Might be better to just not set when empty

Reply:
Reasoning here being that, when no election is going on, we might want to set to 0 (else the metric will sit on the previous observed value between elections, which might give the impression that we're having a bad time)

Reply:
Yeah, still not buying it as that's exactly what the metric is supposed to do: report the last known leader election, and 0 will skew the picture